docs/design/autoware-interfaces/ad-api/list/api/planning/allow_start.md

```diff
@@ -0,0 +1,18 @@
+# /api/planning/allow_start
+
+- Method: function call
+- Type: [autoware_ad_api_msgs/srv/AllowStart](../../../types/autoware_ad_api_msgs/srv/allow_start.md)
+
+## Description
+
+Allow the vehicle to start. This API can be used when the [planning state](./index.md) is STARTING.
```

Comment:
> What does "Allow the vehicle to start" mean?
And what's the difference between the change_to_autonomous (from the STOP state) API?

This API is for starting from all vehicle stops during autonomous driving such as stop signs and obstacles. The operation mode API is for switching between autonomous operation and others (e.g. bus stops).

Reply:
> I guess it might be possible for us to replace this API with a custom API hook (like /api/hooks/pre_start_moving service) before starting moving.

Good point. That's one way to implement this. Needs discussion on suitable method.

Reply:
> Which do you think is better?

I feel the latter one (Autoware calls the HMI service) is better because it's more secure.
If we go with the former one (= start request), we should add an authentication mechanism and block requests from unknown clients.

> when multiple applications want to hook the start, there are two methods: creating a management node or calling all callback from AD API

I think "calling all callback from AD API" is better because if we add a node, the number of points that must be monitored for fail-safe will increase.

Reply:
I think the former is simpler and easier to use. Security should not be considered here. Other APIs such as routing and mode change are available to everyone. It needs the same mechanism, such as an authentication token.

Reply:
Regarding security, you are right. Other APIs should also be considered. So we don't have to consider it here now.
But as I don't feel the former is simpler, let's wait for comments from other developers. If it's okay for others, it's also okay for me.

Reply:
@yabuta @tkhmy @yn-mrse @yukkysaito @mitsudome-r 
Do you have any comments?

Reply:
Since there are no comments, I will implement my method. If it needs to be changed after trying, I will do it in the next version.